File: php/testcode/massassign_031.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_massassign_identity_map_post
function massassign031(BenchmarkRequest $req): BenchmarkResponse {
    $filtered = array_map(fn($v) => $v, $_POST);
    $model = new stdClass();
    foreach ($filtered as $k => $v) {
        $model->$k = $v; // vuln-code-snippet vuln-line php_massassign_identity_map_post
    }
    return BenchmarkResponse::ok('mapped');
}
// vuln-code-snippet end php_massassign_identity_map_post

File: ruby/testcode/massassign_031.rb
require_relative 'shared'

User = Struct.new(:name, :email, :role, :bio, :admin, keyword_init: true)
class << User; def create(h); new(**h.transform_keys(&:to_sym).slice(:name,:email,:role,:bio,:admin)); end; end

def find_user(id)
  User.new(name: 'target', role: 'user')
end

# vuln-code-snippet start ruby_massassign_role_elevation
def handler(req)
  current_user_attrs = req.post_data
  user = find_user(req.param('id'))
  user.update(current_user_attrs) # vuln-code-snippet vuln-line ruby_massassign_role_elevation
  BenchmarkResponse.json({ ok: true })
end
# vuln-code-snippet end ruby_massassign_role_elevation
